start

```diff
@@ -537,6 +537,16 @@ function start_postgres() {
     return 0
   fi
 
+  # Check that postgres can start and load the data successfully. If it can't
+  # successfully do this, then something is in a bad state. For example, a newer
+  # a persistent volume may be mounted with postgres data which is incompatible
+  # with the version of postgres in the container. Log any errors from postgres
+  # to help the operator debug the situation.
+  if ! postgres_logs=$(sudo -u postgres sh -c "echo -n | $PGBIN/postgres --single -E -D $PGDATA -c config_file=$PGHOME/etc/postgresql.conf" 2>&1); then
+    echo "$postgres_logs" | sed -u 's/^/postgres: /'
```

Comment:
These couple lines are a bit of a doozie. I'm using the fact that you can run postgres in single user mode, and if we send some stdin to it, with `echo -n`, it'll cause it to start up, then exit immediately. It starts up, tries to load the data directory, and errors if it can't.